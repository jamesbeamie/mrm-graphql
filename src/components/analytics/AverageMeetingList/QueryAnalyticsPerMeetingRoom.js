import React from 'react';
import PropTypes from 'prop-types';
import timeConvert from '../../helpers/timeConverter';

export const getAverageTime = (time, count) => (
  count === 0 ? 0 : Math.round(time / count)
);


/**
 * Component for QueryAnalyticsPerMeetingRoom
 *
 * @param {Object} data
 *
 * @returns {JSX}
 */
const QueryAnalyticsPerMeetingRoom = ({ data }) =>
  data.meetingDurationAnalytics.map(({ roomName, count, totalDuration }, index) => (
    <div className="table__row--analytics" key={(index + 1).toString()}>
      <span>{roomName}</span>
      <span>{count}</span>
      <span>{timeConvert(getAverageTime(totalDuration, count))}</span>
    </div>
  ));

QueryAnalyticsPerMeetingRoom.propTypes = {
  data: PropTypes.instanceOf(Object).isRequired,
};

export default QueryAnalyticsPerMeetingRoom;
