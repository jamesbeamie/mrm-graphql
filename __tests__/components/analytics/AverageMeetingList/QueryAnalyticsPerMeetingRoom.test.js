import React from 'react';
import { shallow } from 'enzyme';
import QueryAnalyticsPerMeetingRoom from '../../../../src/components/analytics/AverageMeetingList/QueryAnalyticsPerMeetingRoom';

describe('QueryAnalyticsPerMeetingRoom component', () => {
  const props = {
    data: {
      meetingDurationAnalytics: [
        {
          roomName: 'Malabo',
          count: 63,
          totalDuration: 4355,
        },
      ],
    },
  };
  const wrapper = shallow(<QueryAnalyticsPerMeetingRoom {...props} />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
