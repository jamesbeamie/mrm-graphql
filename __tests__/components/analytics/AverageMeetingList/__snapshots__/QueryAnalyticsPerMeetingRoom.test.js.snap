// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryAnalyticsPerMeetingRoom component renders correctly 1`] = `
<div
  className="table__row--analytics"
  key="1"
>
  <span>
    Malabo
  </span>
  <span>
    63
  </span>
  <span>
    1 Hour 9 Minutes
  </span>
</div>
`;
